server/src/lib/routes/verify/get.ts

```diff
@@ -63,6 +64,16 @@ export default function (vars: ServerVariables) {
   return function (req: Express.Request, res: Express.Response)
     : BluebirdPromise<void> {
     let authSession: AuthenticationSession;
+
+    const originalUrl = ObjectPath.get<Express.Request, string>(req, "headers.x-original-url");
+    const domain = DomainExtractor.fromUrl(originalUrl);
+
+    if (vars.accessController.isWhitelisted(domain, req.ip)) {
```

Comment:
You're right about this and can definitely modify this.
We will fix this up shortly.

Reply:
Yep confirmed, this behaviour is correct and using `req.ip` shouldn't be an issue.

Reply:
Yes it failed when I ran the code with docker-compose. req.ip was set with the IP of the frontend nginx, instead of the docker gateway...

Reply:
I was also thinking of using `X-Real-IP` but did not find it in any official doc or RFC...

Reply:
I agree the client could probably spoof X-Forwarded-For unless the frontend proxy flushes it before letting it go with the actual client IP.

Reply:
Hello @nightah , the query is not originating from a Docker container in integration tests right now. The client running the queries is on the host machine and target the port of the frontend nginx through the docker-compose gateway.

Note: all following IPs are masked with /24.

So; the flow for integration tests is 
```
host (running docker engine) (192.168.0.10) -> docker gateway (172.10.0.1) -> nginx frontend (172.10.0.10) -> Authelia (172.10.0.11)
```
So we expect the query to be originating from the docker gateway in that case and req.ip to be 172.10.0.1. But it was not in my tests, I got 172.10.0.10.

In the tests I have in mind the flow will indeed be:
```
host (192.168.0.10) -> docker gateway (172.10.0.1) -> "proxy (172.10.0.20, emulating a client)" -> nginx frontend (172.10.0.10)  -> Authelia (172.10.0.11)
```
So, in that case I'm expecting to see 172.10.0.20 in req.ip. I need to check if it's the case but given what I saw from my latest tests, I don't think it will work.